extensions-core/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidKubernetesClient.java

```diff
@@ -19,21 +19,64 @@
 
 package org.apache.druid.k8s.overlord.common;
 
+import com.google.common.base.Preconditions;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.batch.v1.Job;
 import io.fabric8.kubernetes.client.Config;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import org.apache.druid.error.DruidException;
+import org.apache.druid.java.util.emitter.EmittingLogger;
 import org.apache.druid.k8s.overlord.common.httpclient.DruidKubernetesHttpClientFactory;
 
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
 public class DruidKubernetesClient implements KubernetesClientApi
 {
+  private static final EmittingLogger log = new EmittingLogger(DruidKubernetesClient.class);
+
+  public static final String JOB_NAME_INDEX = "byJobName";
+  public static final String OVERLORD_NAMESPACE_INDEX = "byOverlordNamespace";
+
+  public static final String ENABLE_INFORMERS_KEY = "druid.k8s.informers.enabled";
+  public static final String INFORMER_RESYNC_PERIOD_MS_KEY = "druid.k8s.informers.resyncPeriodMs";
+  private static final long DEFAULT_INFORMER_RESYNC_PERIOD_MS = 300000L; // 5 minutes
+
   private final KubernetesClient kubernetesClient;
+  private final SharedIndexInformer<Pod> podInformer;
+  private final SharedIndexInformer<Job> jobInformer;
```

Comment:
Can we add a `stop()` method that invokes stop on the informers to clean up any resource usage?